### PR TITLE
ocm: added new policy builder and unit tests

### PR DESCRIPTION
**Pull request by klaskosk**
Added `NewPolicyBuilder` as well as some methods to add `PolicyTemplates` and set the `RemediationAction`. Tests for the new methods are included.

### CODE REVIEW

**Comment by kononovn** [comment]
@klaskosk Thanks for your contribution the PR looks good. Could you please add unit tests for the rest of the functions ?

**Comment by klaskosk** [comment]
@kononovn Thanks, I've gone ahead and added the other unit tests

File: pkg/ocm/policy.go [hunk 1]
```diff
@@ -26,10 +26,47 @@ type PolicyBuilder struct {
 	errorMsg string
 }
 
+// NewPolicyBuilder creates a new instance of PolicyBuilder.
+func NewPolicyBuilder(apiClient *clients.Settings, name, nsname string) *PolicyBuilder {
```
Comment (kononovn) [inline pkg/ocm/policy.go:30]: Can you create policy with name and namespaces only? Does this resource have any other required parameter?

File: pkg/ocm/policy.go [hunk 2]
```diff
@@ -179,6 +216,45 @@ func (builder *PolicyBuilder) Update(force bool) (*PolicyBuilder, error) {
 	return builder, err
 }
 
+// WithRemediationAction sets a RemediationAction in the policy definition.
+func (builder *PolicyBuilder) WithRemediationAction(action policiesv1.RemediationAction) *PolicyBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
```
Comment (kononovn) [inline pkg/ocm/policy.go:224]: Please add log, something like:
glog.V(100).Infof("Creating policy %s in namespace %s with RemediationAction: %v",

File: pkg/ocm/policy.go [hunk 3]
```diff
@@ -179,6 +216,45 @@ func (builder *PolicyBuilder) Update(force bool) (*PolicyBuilder, error) {
...
+	return builder
+}
+
+// WithAdditionalPolicyTemplate appends a PolicyTemplate to the PolicyTemplates in the policy definition.
+func (builder *PolicyBuilder) WithAdditionalPolicyTemplate(template *policiesv1.PolicyTemplate) *PolicyBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
```
Comment (kononovn) [inline pkg/ocm/policy.go:244]: Please add log message.

File: pkg/ocm/policy_test.go [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+	for _, testCase := range testCases {
+		_, err := testCase.testBuilder.Delete()
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, testCase.testBuilder.Object)
+		}
+	}
+}
+
```
Comment (kononovn) [inline pkg/ocm/policy_test.go:227]: Please add 1 more test function for Update

**Review by kononovn — CHANGES_REQUESTED** [review]
Overall looks good, just few minor comments to fix.

File: pkg/ocm/policy.go (hunk 1 above)
Comment (klaskosk) [reply pkg/ocm/policy.go:30]: I tried to create a policy and it does look like having at least one policy template is required, so I'll go ahead and add that parameter

**Comment by klaskosk** [comment]
@kononovn Thanks, made the changes

**Review by kononovn — APPROVED** [review]
/lgtm

File: pkg/ocm/policy.go [hunk 5]
```diff
@@ -179,6 +226,49 @@ func (builder *PolicyBuilder) Update(force bool) (*PolicyBuilder, error) {
...
+	glog.V(100).Infof("Setting RemediationAction for policy %s to %v", builder.Definition.Name, action)
+
+	// Lowercase versions are allowed even if there's no constant for them in policiesv1.
+	if action != policiesv1.Inform && action != policiesv1.Enforce && action != "inform" && action != "enforce" {
+		glog.V(100).Info("The RemediationAction to be set in the Policy spec is not 'Inform' or 'Enforce'")
+
+		builder.errorMsg = "remediation action in policy spec must be 'Inform' or 'Enforce'"
```
Comment (achuzhoy) [inline pkg/ocm/policy.go:241]: nit: must be either

**Review by achuzhoy — DISMISSED** [review]

**Review by cdvultur — APPROVED** [review]
/lgtm

**Review by kononovn — APPROVED** [review]

**Pull request merged**